### PR TITLE
Add new compute_diagonal() test

### PR DESCRIPTION
**Pull request by peterrum**
to test different refinement configurations in 2 and 3D

extracted from #12560

### CODE REVIEW

File: tests/matrix_free/compute_diagonal_06.cc [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+  return result;
+}
+
+int
+main(int argc, char **argv)
+{
+  Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
+  MPILogInitAll                    all;
+
```
[the remark this answers is not in the document]
Comment (kronbichler) [reply tests/matrix_free/compute_diagonal_06.cc:135]: The reason I am asking is the serial CI.

**Comment by kronbichler** [comment]
Can you rename the expected test output such that it is not depending on p4est and MPI any more? Then this is good to go.

**Comment by peterrum** [comment]
> Can you rename the expected test output such that it is not depending on p4est and MPI any more? Then this is good to go.

Done!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**